src/geocoding/slf/tests/test_locparser.py

```diff
@@ -0,0 +1,222 @@
+
+    points = list(actual._points())
+    assert len(points) == 2
+    assert points[0].latitude() == 1
+    assert points[0].longitude() == 2
+    assert points[1].latitude() == 3
+    assert points[1].longitude() == 4
+
+    assert [] == list(actual.enum_points())  # already consumed stream above
+
+
+def test_parse_ngsi_polygon():
+    entity = {
+        'location': {
+            'value': ['1, 2', '3,4', '-1,-1', '1,2'],
+            'type': 'geo:polygon'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is not None
+    assert isinstance(actual, SlfPolygon)
+
+    points = list(actual._points())
+    assert len(points) == 4
+    assert points[0].latitude() == 1
+    assert points[0].longitude() == 2
+    assert points[1].latitude() == 3
+    assert points[1].longitude() == 4
+    assert points[2].latitude() == -1
+    assert points[2].longitude() == -1
+    assert points[3].latitude() == 1
+    assert points[3].longitude() == 2
+
+    assert [] == list(actual.enum_points())  # already consumed stream above
+
+
+def test_parse_ngsi_box():
+    entity = {
+        'location': {
+            'value': ['1, 2', '3,4'],
+            'type': 'geo:box'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is not None
+    assert isinstance(actual, SlfBox)
+    assert actual.bottom_right_corner().latitude() == 1
+    assert actual.bottom_right_corner().longitude() == 2
+    assert actual.top_left_corner().latitude() == 3
+    assert actual.top_left_corner().longitude() == 4
+
+
+def test_parse_unknown_location_type():
+    entity = {
+        'location': {
+            'value': ['1, 2', '3,4'],
+            'type': 'geo:???'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is None
+
+
+def test_parse_unsupported_location_value():
+    entity = {
+        'location': {
+            'value': {'start': '1, 2', 'end': '3,4'},
+            'type': 'geo:line'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is None
+
+
+def test_parse_invalid_location_value():
+    entity = {
+        'location': {
+            'value': ['1, 2', '3, wrong!'],
+            'type': 'geo:line'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is None
+
+
+def test_parse_empty_location_value():
+    entity = {
+        'location': {
+            'value': [],
+            'type': 'geo:line'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is None
+
+
+def test_parse_empty_location_coords():
+    entity = {
+        'location': {
+            'value': ['', '1,2'],
+            'type': 'geo:line'
+        }
+    }
+    actual = from_location_attribute(entity)
+
+    assert actual is None
+
+
+def test_parse_no_location():
+    entity = {}
+    actual = from_location_attribute(entity)
+
+    assert actual is None
+
+
+def test_parse_geojson_location():
```

Comment:
How about tests for other types of `geo:json` other than `Point`? 
Shall we move these to a different test module?

Reply:
added more tests :-)